### ISSUE TITLE
Battery Drain on Disconnect

### ISSUE THREAD

**Issue by kmicha19-ford** [smartdevicelink/sdl_ios#1889]
Bug Report

iPhones may experience significant battery drain after disconnecting from Sync 3 or Sync 4 - USB and Bluetooth. This has resulted in one star reviews on the App Store. Apple reports that we are not closing EASessions after disconnect. 

Reproduction Steps

Possible combination of events that may produce the problem.

With AppLink enabled:

1. Close FP App, Connect Phone, Wait for AppLink ready, Disconnect Phone
2. Close FP App, Connect Phone, Open FP AppLink App, Disconnect Phone
3. Close FP App, Connect Phone, Open FP AppLink App, Open FP App, Disconnect Phone
4. Open FP App, Connect Phone, Wait for AppLink ready, Disconnect Phone,
5. Open FP App, Connect Phone, Open FP AppLink App, Disconnect Phone

With CarPlay enabled:

1. Close FP App. Connect Phone, Wait for CarPlay screen to appear, Disconnect Phone
2. Close FP App. Connect Phone, Wait for CarPlay screen to appear, Open some CarPlay Apps, Disconnect Phone
3. Close FP App. Connect Phone, Wait for CarPlay screen to appear, Open some CarPlay Apps, Open FP App, Disconnect Phone
4. Open FP App. Connect Phone, Wait for CarPlay screen to appear, Open some CarPlay Apps, Disconnect Phone

Expected Behavior

Disconnecting the phone should not result in an unexpected battery drain

Observed Behavior

Battery may drain nearly empty in a matter of hours.
Phones may become noticeably warm
Neither closing the app nor rebooting the phone is known to stop the battery drain
Users uninstall the app
Users write one star reviews

OS & Version Information

iOS Version 14.0
SDL iOS Version: 7.0.0
Testing Against: n/a

**Comment by joeljfischer** [comment]
Hi @kmicha19-ford, it looks like this is a duplicate of #1799

**Comment by kmicha19-ford** [comment]
This supercededs 1799 which should be closed.

**Comment by joeljfischer** [comment]
@kshala-ford @thedavidharris are you okay with that? To me it seems like the information contained here should just be a comment on #1799 so that we don't lose the historical context and tracking.

**Comment by kmicha19-ford** [comment]
Some more info. I started working on the battery drain problem in late November, as fresh set of eyes on the battery drain problem. The end result is a refactor of code in the SDL with OEM Integrations repo used by ACVL. This refactor is now in FordPass release 3.14.  I created this Battery Drain on Disconnect on the public repo so I can create a PR with this same code changes. The PR comments and code will detail the code changes.  The code mention in 1799 wil be gone with this refactor.  So closing 1799 is a suggestion but that's all.
